src/agent.rs

```diff
@@ -367,59 +399,85 @@ impl AgentThread {
         debug!("handle request");
         let response = match request.kind {
             RequestKind::AllocMR(param) => {
+                // TODO: error handling
                 let mr = self.inner.allocator.alloc(
                     &Layout::from_size_align(param.size, param.align)
                         .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?,
                 )?;
-                let token = mr.token();
-                let response = AllocMRResponse { token };
+                let token = mr.token_with_timeout(param.timeout).map_or_else(
+                    || {
+                        Err(io::Error::new(
+                            io::ErrorKind::InvalidInput,
+                            "wrong timeout value",
+                        ))
+                    },
+                    Ok,
+                )?;
                 // the MR is newly created, it's impossible to find it in the map
-                let _old = self.inner.mr_own.lock().await.insert(token, mr);
+                self.inner
+                    .rmr_manager
+                    .record_mr(token, mr, param.timeout)
+                    .await
+                    .map_or_else(
+                        |err| error!("{:?}", err),
```

Comment:
How to handle the error? Error logging is not enough.

Reply:
Actually it's unreachable